### ISSUE TITLE
Upgrade Moped RDS instance to use Postgres 16

### ISSUE THREAD

**Issue by mddilley** [cityofaustin/atd-data-tech#17924]
In https://github.com/cityofaustin/atd-data-tech/issues/17456, Frank aligned our local Postgis version to match our production instance and documented the following steps to upgrade our Postgres version. **EOL of Postgres 12 is 11/14/2024.**

**In Scope**
- [ ] Snapshot the DB
- [ ] Start up a throw-away RDS instance based off that snapshot. This is our guinea pig.
- [ ] **Target PG version 16 and check RDS docs to confirm if the upgrade path is direct or via an intermediate version.**
- [ ] Edit the RDS configuration to the target PG version, and ask it to do the migration now instead of waiting for a maintenance window.
- [ ] Observe if it works
- [ ] Discard the throw-away RDS instance
- [ ] Schedule downtime (or do during Moped release maintenance window?)
- [ ] Back-up the database
- [ ] Do it for real with the real RDS instance

**Resources**
- [Frank's thread](https://austininnovation.slack.com/archives/CNUEPKLB1/p1716407327890799)
- [Postgres 12 EOL is 11/14/2024](https://endoflife.date/postgresql)
- https://www.enterprisedb.com/blog/performance-comparison-major-PostgreSQL-versions
- for reference, the VZ Aurora instance is Postgres 14.7

**Heads up**
This is what we face if we do not update. h/t Frank
> Your PostgreSQL 12 databases are approaching end of standard support, and will be charged for RDS Extended Support starting March 1, 2025. To avoid the additional charges due to RDS Extended Support, we recommend upgrading your databases to a newer major engine version before February 28, 2025. To learn more about the RDS Extended Support pricing, refer to the pricing page.

**Comment by mddilley** [comment]
@frankhereford shared in dev sync that Postgres is now at 17! https://www.postgresql.org/about/news/postgresql-17-released-2936/

**Comment by frankhereford** [comment]
After the RDS upgrade failure, @mddilley  surfaced this really helpful documentation that I want to link: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_UpgradeDBInstance.PostgreSQL.MajorVersion.Process.html

**Comment by frankhereford** [comment]
@mddilley and I were able to do a stepwise upgrade from one major version to the next. We had to destroy and recreate the read replica, which we did and updated the CNAME Route 53 record to point at the new read replica endpoint provided by AWS. After the DNS propagated, we were able to confirm that we both could connect to the RR, and this is done. The upgrade process took about 1 and a half hours.
